src/server/cray/cfs/api/controllers/configurations.py

```diff
@@ -192,6 +235,27 @@ def put_configuration_v3(configuration_id, drop_branches=False):
             status=400, title="Error parsing the data provided.",
             detail=str(err))
 
+    # If the put request comes from a specific tenant, make note of it in the record -- we're going to use it in
+    # subsequent data puts and permission checks.
+    requesting_tenant = get_tenant_from_header() or None
+
+    # If the configuration already exists, and the tenant is not owned by the requesting put tenant, then we cannot
+    # allow them to overwrite the existing data for this key.
```

Comment:
Was it a deliberate choice to deviate from the BOS scheme here, and not make the tenant part of the database key itself? I thought in the case of BOS, part of the reason this was done was to avoid letting one tenant get knowledge about resources owned by the other tenant. With this current CFS implementation, tenants can learn that another tenant has a configuration of a particular name. To me, that doesn't seem like a big deal. Especially since our multi-tenancy implementation is quite soft overall. :) But I figured I'd mention it just because this will make CFS inconsistent with BOS in this regard.

Reply:
Yes. Due to the nature of configuration management changes, the global administrator is still on the hook to prevent tenant administrators from doing anything nefarious, like writing an ansible task that delegates_to: <a node they do not own>.